webtool/frontend/src/components/configure.tsx

```diff
@@ -24,63 +24,125 @@ export const Configure: FunctionComponent<{ pilot: Pilot, setPilot: (pilot: Pilo
+            delete: (asset: AssetType) => setPilot(pilot.remove(asset as HouseholdGroup)),
+        },
+        SolarFarm: {
+            display: (show: boolean) => setShowAddSolarFarm(show),
+            save: (asset: AssetType) => setPilot(pilot.create(asset as SolarFarm)),
+            delete: (asset: AssetType) => setPilot(pilot.remove(asset as SolarFarm)),
+        },
+        WindFarm: {
+            display: (show: boolean) => setShowAddWindFarm(show),
+            save: (asset: AssetType) => setPilot(pilot.create(asset as WindFarm)),
+            delete: (asset: AssetType) => setPilot(pilot.remove(asset as WindFarm)),
+        },
+        Battery: {
+            display: (show: boolean) => setShowAddBattery(show),
+            save: (asset: AssetType) => setPilot(pilot.create(asset as Battery)),
+            delete: (asset: AssetType) => setPilot(pilot.remove(asset as Battery)),
+        },
+        HeatStorage: {
+            display: (show: boolean) => setShowAddHeatStorage(show),
+            save: (asset: AssetType) => setPilot(pilot.create(asset as HeatStorage)),
+            delete: (asset: AssetType) => setPilot(pilot.remove(asset as HeatStorage)),
+        },
+    };
+   
+    // Generic save function
+    const saveAsset = (asset: AssetType) => {
+        const type = asset.constructor.name;
+        const handler = assetHandlers[type]?.save;
+        
+        if (handler) {
+            handler(asset);
+        } else {
+            console.error(`No save handler found for type: ${type}`);
+        }
+    };
 
-    const saveBattery = (battery: Battery) => {
-        setPilot(pilot.withBattery(battery))
-    }
+    // Generic delete function
+    const deleteAsset = (asset: AssetType) => {
+        const type = asset.constructor.name;
+        const handler = assetHandlers[type]?.delete;
 
-    const saveHeatStorage = (heatStorage: HeatStorage) => {
-        setPilot(pilot.withHeatStorage(heatStorage))
-    }
+        if (handler) {
+            handler(asset);
+        } else {
+            console.error(`No delete handler found for type: ${type}`);
+        }
+    };
+
+    const handleDisplay = (type: string, show: boolean) => {
+        const handler = assetHandlers[type]?.display;
+
+        if (handler) {
+            handler(show);
+        } else {
+            console.error(`No delete handler found for type: ${type}`);
+        }
+    };
 
     return (
         <Grid gap="2" pt="4">
             {pilot.householdGroups.asJsReadonlyArrayView().map((it, i) =>
-                <HouseholdDisplay key={"householdGroup_" + i} householdGroup={it}/>)}
-            {showAddHouseholdGroup &&
-                <HouseholdForm saveHouseholdGroup={saveHouseholdGroup} hide={() => setShowAddHouseholdGroup(false)}/>}
-
+                <HouseholdDisplay key={"householdGroup_" + i} householdGroup={it} 
+                    toDelete={() => deleteAsset(it)}
```

Comment:
You could do `setPilot(pilot.removeHouseholdGroup(it))`

Seems a bit overengineered.